api/swagger.yaml

```diff
@@ -181,132 +177,168 @@ paths:
                 $ref: '#/components/schemas/Error'
+
 components:
   schemas:
     Command:
       required:
       - type
-      type: object
       properties:
         type:
           type: string
-      discriminator:
-        propertyName: Type_
-      oneOf:
-      - $ref: '#/components/schemas/MoveCommand'
-      - $ref: '#/components/schemas/StopCommand'
-      - $ref: '#/components/schemas/ConstructCommand'
-      - $ref: '#/components/schemas/DecommissionCommand'
-      - $ref: '#/components/schemas/AttackCommand'
-      - $ref: '#/components/schemas/TradeCommand'
-      - $ref: '#/components/schemas/RenameCommand'
+        target:
+          type: integer
+          format: int64
+          nullable: true
+        destination:
+          $ref: "#/components/schemas/Destination"
+        resource:
+          type: integer
+          format: int64
+          nullable: true
+        amount:
+          type: integer
+          format: int64
+          nullable: true
+        ship-class:
+          type: integer
+          format: int64
+          nullable: true
+        name:
+          type: string
+          nullable: true
+
     MoveCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
-        - where
-        type: object
-        properties:
-          destination:
-            $ref: "#/components/schemas/Destination"
+      type: object
+      properties:
+        type:
+          type: string
+        destination:
+          $ref: "#/components/schemas/Destination"
+      required:
+        - type
+        - destination
+
     StopCommand:
-      $ref: '#/components/schemas/Command'
+      type: object
 
     ConstructCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
-        - class
-        type: object
-        properties:
-          class:
-            type: string
-          name:
-            type: string
+      type: object
+      properties:
+        ship-class:
+          type: integer
+          format: int64
+          nullable: true
+        type:
+          type: string
+      required:
+        - type
+        - ship-class
 
     DecommissionCommand:
-      $ref: '#/components/schemas/Command'
+      type: object
+      properties: 
+        type:
+          type: string
+      required:
+        - type
 
     RenameCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
+      type: object
+      properties:
+        type:
+          type: string
+        name:
+          type: string
+          nullable: true
+      required:
         - name
-        type: object
-        properties:
-          name:
-            type: string
+        - type
+
     AttackCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
+      type: object
+      properties:
+        type:
+          type: string
+        target:
+          type: integer
+          format: int64
+          nullable: true
```

Comment:
why nullable?

Reply:
is it related to #53 ? if so, maybe we can guarantee that ids are greater than zero and be done with it?

Reply:
https://github.com/gdg-garage/space-tycoon/pull/57#issuecomment-1207249559

it's not only about IDs. For example, bool values set to false would also be treated as missing. Using pointers is the correct way.

Reply:
So why add nullable to ids? Would it be sufficient to add to booleans and such only?

Reply:
is there a reason to have it inconsistent? Optional values are pointers, and non-optional are values. Simple